### PR TITLE
Add bqetl_fxa_events DAG

### PR DESCRIPTION
**Pull request by jklukas**
Replaces the fxa_events DAG in telemetry-airflow: https://github.com/mozilla/telemetry-airflow/pull/1092

### CODE REVIEW

File: sql/firefox_accounts_derived/fxa_auth_bounce_events_v1/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+friendly_name: FxA Auth Bounce Events
+description: Selected Amplitude events extracted from FxA auth_bounce server logs
+owners:
+  - jklukas@mozilla.com
+labels:
+  application: fxa
+  incremental: true
+  schedule: daily
+scheduling:
+  dag_name: bqetl_fxa_events
```
Comment (scholtzan) [inline sql/firefox_accounts_derived/fxa_auth_bounce_events_v1/metadata.yaml:10]: I guess we'll need to keep the arguments: https://github.com/mozilla/telemetry-airflow/blob/caa17a72b905adc8e975d9310814032855abea87/dags/fxa_events.py#L38

```suggestion
  dag_name: bqetl_fxa_events
  arguments: ['--schema_update_option=ALLOW_FIELD_ADDITION']
```

File: sql/firefox_accounts_derived/fxa_auth_events_v1/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+friendly_name: FxA Auth Events
+description: Selected Amplitude events extracted from FxA auth server logs
+owners:
+  - jklukas@mozilla.com
+labels:
+  application: fxa
+  incremental: true
+  schedule: daily
+scheduling:
+  dag_name: bqetl_fxa_events
```
Comment (scholtzan) [inline sql/firefox_accounts_derived/fxa_auth_events_v1/metadata.yaml:10]: https://github.com/mozilla/telemetry-airflow/blob/caa17a72b905adc8e975d9310814032855abea87/dags/fxa_events.py#L30
```suggestion
  dag_name: bqetl_fxa_events
  arguments: ['--schema_update_option=ALLOW_FIELD_ADDITION']
```

File: sql/firefox_accounts_derived/fxa_content_events_v1/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+friendly_name: FxA Content Events
+description: Selected Amplitude events extracted from FxA content server logs
+owners:
+  - jklukas@mozilla.com
+labels:
+  application: fxa
+  incremental: true
+  schedule: daily
+scheduling:
+  dag_name: bqetl_fxa_events
```
Comment (scholtzan) [inline sql/firefox_accounts_derived/fxa_content_events_v1/metadata.yaml:10]: https://github.com/mozilla/telemetry-airflow/blob/caa17a72b905adc8e975d9310814032855abea87/dags/fxa_events.py#L46
```suggestion
  dag_name: bqetl_fxa_events
  arguments: ['--schema_update_option=ALLOW_FIELD_ADDITION']
```

File: sql/firefox_accounts_derived/fxa_users_last_seen_v1/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+description: Usage aggregations per FxA user per day over a 28-day window
+owners:
+  - jklukas@mozilla.com
+labels:
+  application: fxa
+  incremental: true
+  schedule: daily
+scheduling:
+  dag_name: bqetl_fxa_events
+  depends_on_past: true
```
Comment (scholtzan) [inline sql/firefox_accounts_derived/fxa_users_last_seen_v1/metadata.yaml:11]: Not sure if this is required (see https://github.com/mozilla/telemetry-airflow/blob/caa17a72b905adc8e975d9310814032855abea87/dags/fxa_events.py#L103)

```suggestion
  depends_on_past: true
  start_date: '2019-04-23'
```

File: sql/firefox_accounts_derived/fxa_users_services_first_seen_v1/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  incremental: true
+  schedule: daily
+scheduling:
+  dag_name: bqetl_fxa_events
+  # This query updates the entire existing table every day rather than appending
+  # a new partition, so we need to disable date_partition_parameter and instead
+  # pass submission_date as a generic param.
+  date_partition_parameter: null
+  parameters: ["submission_date:DATE:{{ds}}"]
+  sql_file_path: sql/firefox_accounts_derived/fxa_users_services_first_seen_v1/init.sql
```
Comment (scholtzan) [inline sql/firefox_accounts_derived/fxa_users_services_first_seen_v1/metadata.yaml:16]: I think we might need to add some extra support for having `destination_table` as `None`? https://github.com/mozilla/telemetry-airflow/blob/caa17a72b905adc8e975d9310814032855abea87/dags/fxa_events.py#L157

File: sql/firefox_accounts_derived/fxa_users_services_last_seen_v1/metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+description: Usage aggregations per FxA user per FxA service per day over a 28-day window
+owners:
+  - jklukas@mozilla.com
+labels:
+  application: fxa
+  incremental: true
+  schedule: daily
+scheduling:
+  dag_name: bqetl_fxa_events
+  depends_on_past: true
```
Comment (scholtzan) [inline sql/firefox_accounts_derived/fxa_users_services_last_seen_v1/metadata.yaml:11]:
```suggestion
  depends_on_past: true
  start_date: '2019-10-08'
```

https://github.com/mozilla/telemetry-airflow/blob/caa17a72b905adc8e975d9310814032855abea87/dags/fxa_events.py#L170

File: sql/firefox_accounts_derived/fxa_users_services_first_seen_v1/metadata.yaml (hunk 5 above)
Comment (jklukas) [reply sql/firefox_accounts_derived/fxa_users_services_first_seen_v1/metadata.yaml:16]: Good catch. I will look into it.

File: sql/firefox_accounts_derived/fxa_users_services_first_seen_v1/metadata.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+labels:
+  application: fxa
+  incremental: true
+  schedule: daily
+scheduling:
+  dag_name: bqetl_fxa_events
+  # We completely recreate this table every day;
+  # making it incremental is possible but nuanced since it windows over
+  # events that may cross the midnight boundary.
+  date_partition_parameter: null
```
Comment (jklukas) [inline sql/firefox_accounts_derived/fxa_users_services_first_seen_v1/metadata.yaml:14]: We now sidestep the destination problem by copying the `init.sql` logic into `query.sql` so the execution method is more in line with other queries. This results in code duplication, but that's the case to some extent with existing practice for other `init.sql` queries.

**Pull request merged**